### ISSUE TITLE
'NoneType' object has no attribute 'waitShutdown'

### ISSUE THREAD

**Issue by tardyp** [buildbot/buildbot#6064]
seen on metabuildbot twisted.log

```
        Traceback (most recent call last):
          File "/home/bbuser/venv/lib/python3.6/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
            result = g.send(result)
          File "/usr/home/bbuser/buildbot/master/buildbot/worker/base.py", line 492, in detached
            self._handle_conn_shutdown_notifier(conn)
          File "/home/bbuser/venv/lib/python3.6/site-packages/twisted/internet/defer.py", line 1613, in unwindGenerator
            return _cancellableInlineCallbacks(gen)
          File "/home/bbuser/venv/lib/python3.6/site-packages/twisted/internet/defer.py", line 1529, in _cancellableInlineCallbacks
            _inlineCallbacks(None, g, status)
        --- <exception caught here> ---
          File "/home/bbuser/venv/lib/python3.6/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
            result = g.send(result)
          File "/usr/home/bbuser/buildbot/master/buildbot/worker/base.py", line 484, in _handle_conn_shutdown_notifier
            yield conn.waitShutdown()
        builtins.AttributeError: 'NoneType' object has no attribute 'waitShutdown'
```

**Comment by seankelly** [comment]
I saw this same error in #6054 but after the error I included in that log.

**Comment by p12tic** [comment]
I believe the issue has been fixed in #6056. The last error was on 2021-05-21 and not seen since.
